JumpscaleLibs/sal/reservation_chatflow/reservation_chatflow.py

```diff
@@ -164,14 +164,28 @@ def _nodes_distribute(self, number_of_nodes, farm_names):
                 names_pointer = 0
         return nodes_distribution
 
-    def nodes_filter(self, nodes, free_to_use):
+    def nodes_filter(self, nodes, free_to_use, ip_version=None):
         nodes = filter(j.sal.zosv2.nodes_finder.filter_is_up, nodes)
         nodes = list(nodes)
         if free_to_use:
             nodes = list(nodes)
             nodes = filter(j.sal.zosv2.nodes_finder.filter_is_free_to_use, nodes)
         elif not free_to_use:
             nodes = list(nodes)
+
+        if ip_version:
+
+            use_ipv4 = ip_version == "IPv4"
+
+            if use_ipv4:
+                nodefilter = j.sal.zosv2.nodes_finder.filter_public_ip4
+            else:
+                nodefilter = j.sal.zosv2.nodes_finder.filter_public_ip6
+
+            nodes = filter(j.sal.zosv2.nodes_finder.filter_is_up, filter(nodefilter, nodes))
```

Comment:
This check is already done here https://github.com/threefoldtech/jumpscaleX_libs/pull/230/commits/9ddccea6bdf3ad1a50a04f34b71d530c8d2f19a6#diff-47ef217232882edb91c5726cc77e9c7cR185